### ISSUE TITLE
Upgrade from 5.34.1 to 5.34.2 broke permissions and pg password?

### ISSUE THREAD

**Issue by jayenashar** [mattermost/mattermost-server#17532]
#### Summary
Upgrade from 5.34.1 to 5.34.2 broke permissions and pg password?

#### Steps to reproduce
Install mattermost omnibus 5.30.1 and upgrade incrementally to 5.34.2

#### Expected behavior
mattermost should start

#### Observed behavior (that appears unintentional) 
first error: `Error: failed to load configuration: failed to create store: unable to load on store creation: failed to persist: failed to write file: open /opt/mattermost/config/config.json: permission denied`

checked perms:
```
ubuntu@ip-172-26-3-183:/var/log$ ls -l /opt/mattermost/config/
total 28
-rw-r--r-- 1 root       root         243 Apr 16 18:47 README.md
-rw-r--r-- 1 root       root        1014 Apr 16 18:47 cloud_defaults.json
-rw-r--r-- 1 root       root       20040 Apr 16 18:47 config.json
```

upgraded from 5.33.3 to 5.34.0 on 2021-04-16 06:45:30 and to 5.34.1 on 2021-04-18 06:26:31 .  not sure where Apr 16 18:47 comes from

changed perms: `sudo chown mattermost: config.json`

second error: `pq: password authentication failed for user "mmuser"`

i don't know what to do.  i don't recall configuring postgres.

**Comment by agnivade** [comment]
It looks like your config.json has got reset to the default values, which use the `mmuser` user. I'd perhaps suggest to take a look at the file and rectify the database credentials.

**Comment by jayenashar** [comment]
i did this but now it's trying to connect to mattermost_test.  i wish i backed up the config file...

```
ubuntu@ip-172-26-3-183:/opt/mattermost/config$ sudo -u postgres psql mattermost
psql (13.2 (Ubuntu 13.2-1.pgdg20.04+1))
Type "help" for help.

mattermost=# \password mmuser
Enter new password: 
Enter it again: 
mattermost=# \q
```

**Comment by agnivade** [comment]
You need to update the config to whatever database you want to connect to.

**Comment by jayenashar** [comment]
i found a backup but it has the same password and db name.  something else has gone wrong since 5.34.1 worked fine with this connection string even though the database `mattermost_test` doesn't exist.

**Comment by jayenashar** [comment]
found the real settings in /etc/mattermost/mmomni.mattermost.env .  will change the mmuser password back to that and 🤞🏾

**Comment by jayenashar** [comment]
fixed it by downgrading to 5.34.1 and reverting the db password.  had some [i believe unrelated] issues with cerbot.  will wait for 5.34.3 or 5.35 before trying to upgrade again

**Comment by amyblais** [comment]
I'll close this for now as it's an older issue, but feel free to re-open it if the issue persists on the latest version,
